pkg/manager/member/tidb_member_manager.go

```diff
@@ -551,16 +555,34 @@ func getNewTiDBSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap)
 		{Name: "config", ReadOnly: true, MountPath: "/etc/tidb"},
 		{Name: "startup-script", ReadOnly: true, MountPath: "/usr/local/bin"},
 	}
-	if tc.IsTLSClusterEnabled() {
+	if tc.IsTLSClusterEnabled() && !tc.IsHeterogeneous() {
 		volMounts = append(volMounts, corev1.VolumeMount{
 			Name: "tidb-tls", ReadOnly: true, MountPath: clusterCertPath,
 		})
 	}
-	if tc.Spec.TiDB.IsTLSClientEnabled() {
+	if tc.Spec.TiDB.IsTLSClientEnabled() && !tc.IsHeterogeneous() {
 		volMounts = append(volMounts, corev1.VolumeMount{
 			Name: "tidb-server-tls", ReadOnly: true, MountPath: serverCertPath,
 		})
 	}
+	if tc.IsHeterogeneous() {
```

Comment:
I think we can combine this block with that from L627, no need to get the tc twice.